File: dotnet/src/Iris/Frontend.Workbench/src/PanelLeft.js
import * as React from "react";
import css from "../css/PanelLeft.less";

function cardClicked(title) {
  console.log(title);
}

const Card = props => (
  <div className="iris-panel-left-child" onClick={() => cardClicked(props.title)}>
    <div>{props.letter}</div>
    <div>
      <p><strong>{props.title}</strong></p>
      <p>{props.text}</p>
    </div>
  </div>
);

export default class PanelLeft extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="iris-panel-left">
        <Card key={0} letter="P" title="Project Overview (Big)" text="Cluster Settings" />
        <Card key={1} letter="P" title="Project Overview (Small)" text="Cluster Settings" />
        <Card key={2} letter="B" title="Branches" text="Cluster Settings" />
        <Card key={3} letter="L" title="LOG" text="Cluster Settings" />
        <Card key={4} letter="U" title="User Management" text="Cluster Settings" />
        <Card key={5} letter="H" title="Unassigned Hosts" text="Cluster Settings" />
        <Card key={6} letter="R" title="Remotter" text="Cluster Settings" />
        <Card key={7} letter="S" title="Project Settings" text="Cluster Settings" />
        <Card key={8} letter="L" title="Library" text="Graph View" />
      </div>
    )
  }
}

File: dotnet/src/Frontend/fable/Frontend/Elmish/PanelLeft.fs
module Iris.Web.PanelLeft

open Fable.Helpers.React
open Fable.Helpers.React.Props
open Fable.Core.JsInterop
open Elmish.React
open Helpers

let onClick id _ =
    printfn "PanelLeft clicked %i" id

let card key letter title text =
    div [
        Key (string key)
        Class "iris-panel-left-child"
        OnClick (onClick key)
    ] [
        div [] [str letter]
        div [] [
            p [] [strong [] [str title]]
            p [] [str text]
        ]
    ]

let render () =
    div [Class "iris-panel-left"] [
        card 0 "L" "LOG" "Cluster Settings"
        card 1 "G" "Graph View" "Cluster Settings"
        card 2 "C" "Cue Player" "Cluster Settings"
        card 3 "P" "Project View" "Cluster Settings"
        card 4 "T" "Test Widget" "Cluster Settings"
        card 5 "R" "Cluster" "Cluster Settings"
        card 6 "D" "Discovery" "Cluster Settings"
        card 7 "H" "Unassigned Hosts" "Cluster Settings"
        card 8 "R" "Remotter" "Cluster Settings"
        card 9 "S" "Project Settings" "Cluster Settings"
        card 10 "L" "Library" "Graph View"
        card 11 "P" "Project Overview (Big)" "Cluster Settings"
    ]

let view () =
  lazyViewWith
    (fun x y -> obj.ReferenceEquals(x, y))
    (fun () -> render())
    ()
